src/scripts/generate_binary_ops.py

```diff
@@ -0,0 +1,320 @@
+        types_count = 0 # Number of calculation types needed
+
+    # Initialize the generator
+    def __init__(self, options):
+        self.num_operands = options.operand_count # Number of operands in the expression
+        self.options = options
+
+    # Adding functions to the library and generating the expression with operands and operators
+    def expression_generator(self,calc):
+        self.functions_lib(calc)
+        self.expression_maker(calc)
+
+    # Adding functions to the library
+    def functions_lib(self, calc):
+        self.operators_list = [] # List of currently used operators
+        self.oper_funcs = {
+              "add": self.add,
+              "sub": self.sub,
+              "mul": self.mul,
+              "div": self.div,
+              "mod": self.mod,
+              "and": self.bitand,
+              "xor": self.bitxor,
+              "or": self.bitor,
+              "leftshift": self.bitleftshift,
+              "rightshift": self.bitrightshift,
+              "unsignedrightshift": self.bitunsignedrightshift
+        } # Functions for the operators
+
+    # Generating the expression with operands and operators
+    def expression_maker(self, calc):
+        for op in self.options.test_cases:
+            if calc == self._InnerValues.arith:
+                if op in self._InnerValues.arithmetic_operators:
+                    self.operators_list.append(self.oper_funcs[op])
+            elif calc == self._InnerValues.bit:
+                if op in self._InnerValues.bitwise_operators:
+                    self.operators_list.append(self.oper_funcs[op])
+            else:
+                if op in self._InnerValues.uint32_operators:
+                    self.operators_list.append(self.oper_funcs[op])
+
+        self.operands = [0] * self.num_operands # Creating a list for operands according to number of operands
+        self.operators = [0] * (self.num_operands - 1) # Creating a list for operators according to number of operands
+
+        # Setting max and min according to the calculations
+        maxnum = self.set_min_max(calc)["max"]
+        minnum = self.set_min_max(calc)["min"]
+
+        # Generating the expression
+        for i in range(self.num_operands - 1):
+            self.operands[i] = random.randint(minnum, maxnum)
+            self.operators[i] = random.randint(0, len(self.operators_list) - 1)
+        self.operands[self.num_operands - 1] = random.randint(minnum, maxnum)
+
+    # Arithmetic operations
+
+    # Addition
+    def add(self, op1, op2):
+        sum = op1 + op2
+        if sum < _constants.min: # Minimum / maximum number to safely add to operand 1
+            op2 = _constants.min - random.randint(_constants.min, op1)
+            sum = op1 + op2
+        elif sum > _constants.max:
+            op2 = _constants.max - random.randint(op1, _constants.max)
+            sum = op1 + op2
+        return {'oper': '+', 'op2': op2, 'sum': sum}
```

Comment:
How about creating a new function like:

``` py
def create_operation_result_object (oper, op2, sum):
  return {'oper': oper 'op2': op2, 'sum': sum}
```

After that each arithmetic/binary operations could return with:

``` py
return create_operation_result_object ('+,-,*,/, etc.', op2, sum):
```